### PR TITLE
fix Hardware Serial for esp32

### PR DESCRIPTION
**Pull request by nguyenhuudamy**

### CODE REVIEW

**Comment by mandulaj** [comment]
Hi @nguyenhuudamy Thanks for the PR. However I don't like the idea of changing an established API. It would for sure break my projects not to mention anyone who already uses this library... Perhaps we can overload the function or wrap it in a #if #else to fix it just for the esp32 platform while retaining the Arduino compatible API.

**Comment by ManuelJimenezBuendia** [comment]
What about something like this?
**.h**
#ifdef ESP32
	PZEM004Tv30(HardwareSerial* port, int rx_pin, int tx_pin, uint8_t addr=PZEM_DEFAULT_ADDR);
#endif

**.cpp**
#ifdef ESP32
  PZEM004Tv30::PZEM004Tv30(HardwareSerial* port, int rx_pin, int tx_pin, uint8_t addr)
  {
    port->begin(PZEM_BAUD_RATE, SERIAL_8N1, rx_pin, tx_pin);  //Baud rate, parity mode, RX, TX
    this->_serial = port;
    this->_isSoft = false;
    init(addr);
  }
 #endif

File: PZEM004Tv30.h [hunk 1]
```diff
@@ -62,7 +62,7 @@ class PZEM004Tv30
 #if defined(PZEM004_SOFTSERIAL)
     PZEM004Tv30(uint8_t receivePin, uint8_t transmitPin, uint8_t addr=PZEM_DEFAULT_ADDR);
 #endif
-    PZEM004Tv30(HardwareSerial* port, uint8_t addr=PZEM_DEFAULT_ADDR);
+    PZEM004Tv30(HardwareSerial* port, uint8_t receivePin, uint8_t transmitPin, uint8_t addr=PZEM_DEFAULT_ADDR);
```
Comment (ManuelJimenezBuendia) [inline PZEM004Tv30.h:65]: Add #ifdef ESP32 ..... #endif fro ESP32? And don't remove previous constructor

**Review by mandulaj — CHANGES_REQUESTED** [review]
So I added the requested `#if def` and modified the examples in order to be compatible. It has compiled however I have no way of testing it on any board at the moment. Could anyone with access to an ESP32 board try this? Maybe @ASMotionLab if you have time.

**Comment by ASMotionLab** [comment]
Thanks @mandulaj, looks great at a glance - I will check it out this evening, test with my ESP32 and report 👍

**Comment by mandulaj** [comment]
I managed to get hold of an ESP32 and it seems to work fine... So I am ready to merge right away.

**Review by mandulaj — APPROVED** [review]
I added the ESP32 switch and it works fine now

**Pull request merged**